app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -87,5 +91,21 @@ def two_factor_options_form_params
     def in_ab_test_bucket?
       ab_test_bucket(:DESKTOP_FT_UNLOCK_SETUP) == (:desktop_ft_unlock_option_shown)
     end
+
+
+    def threatmetrix_variables
+      return {} unless FeatureManagement.account_creation_device_profiling_collecting_enabled?
+      session_id = generate_threatmetrix_session_id
+
+      {
+        threatmetrix_session_id: session_id,
+        threatmetrix_javascript_urls: threatmetrix_javascript_urls(session_id),
+        threatmetrix_iframe_url: threatmetrix_iframe_url(session_id),
+      }
+    end
+
+    def generate_threatmetrix_session_id
+      session[:threatmetrix_session_id] ||= SecureRandom.uuid
```

Comment:
Did we use `session` originally because they're not yet signed in, and if they're now signed-in would we want to use `user_session`? I think if we did, we'll want to be aware of whether we want this to be distinct from the IdV session value, and rename accordingly (might be wise to have distinct names anyways).